### PR TITLE
fake data generator notebook

### PR DESCRIPTION
**Pull request by jakemulf**

### CODE REVIEW

**Comment by jakemulf** [comment]
Blocked by https://github.com/deephaven/deephaven-core/issues/1845

**Review by hythloda — COMMENTED** [review]
I love the mentality behind this notebook. I hope the nexts ones do something with the data. More descriptive table names would be ideal. Also not too much WOW so might want to increase table sizes? Perhaps more data, have not ran it to see if that would be okay time wise.

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Create tables and fake data
+
+In this demo, we show how to perform real-time operations on tables. To start off, we will need to create some tables with data in them.
+
+The following Python code contains a method that creates a table with random integers and characters in it, with each row in the table containing a time-stamp as well.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:5]:
```suggestion
The following Python code contains a method that creates a table of random integers and characters, with each row in the table also containing a timestamp.
```

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+from deephaven.DateTimeUtils import plus
+import deephaven.Types as dht
+
+import random
+import string
+
+def create_random_table(number_of_rows, start_time, time_offset):
+    """
+    Creates a Deephaven table containing rows of random integers from 1 to 99, random
+    uppercase characters, and time-stamps.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:18]:
```suggestion
    uppercase characters, and timestamps.
```

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    
+    Parameters:
+        number_of_rows (int): The number of rows that the resulting table will contain.
+        start_time (DateTime): The Deephaven date-time of the first row in the table.
+        time_offset (Period): A Period object representing the time-stamp difference between
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:23]:
```suggestion
        time_offset (Period): A Period object representing the timestamp difference between
```

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    for i in range(number_of_rows):
+        random_number = random.randint(1, 100)
+        random_character = random.choice(string.ascii_uppercase)
+        table_writer.logRow(time, random_number, random_character)
+        time = plus(time, time_offset)
+
+    return table_writer.getTable()
+```
+
+We can use this method to create some tables with random data in it.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:42]:
```suggestion
We can use this method to create some tables with random data.
```

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+time_offset = Period("T10S")
+t3 = create_random_table(5, start_time, time_offset)
+t4 = create_random_table(50, start_time, time_offset)
+
+time_offset = Period("T0.1S")
+t5 = create_random_table(5, start_time, time_offset)
+t6 = create_random_table(50, start_time, time_offset)
+```
+
+[The next notebook](A2%20Filter%20and%20decorate.md) will show how to filter and decorate this data.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:62]: doesn't exit yet, though?

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# Create tables and fake data
+
+In this demo, we show how to perform real-time operations on tables. To start off, we will need to create some tables with data in them.
```
[the remark this answers is not in the document]
Comment (margaretkennedy) [reply demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:3]: After reading the whole thing, the purpose of this notebook is to generate some fake data that we can manipulate later. 

"Throughout this demo notebook series, we show many of the ways to interact with real-time data in Deephaven. Here, we create some tables with fake data; in other notebooks, we show how to perform table operations on that data. Knowing how to create fake ticking tables can be useful for many applications. For instance, you may want to..."

@jakemulf I don't know the best way to finish that sentence. Timetables are useful for lots of things, but those reasons don't necessarily apply here.

[the remark this answers is not in the document]
Comment (jjbrosnan) [reply demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:3]: I haven't read the whole thing yet, but if the notebook is titled "create tables and fake data" then it should focus on creating the data, not the operations on the data.  The first line conflicts with the title of the notebook.

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+t1 = create_random_table(5, start_time, time_offset)
+t2 = create_random_table(50, start_time, time_offset)
+
+time_offset = Period("T10S")
+t3 = create_random_table(5, start_time, time_offset)
+t4 = create_random_table(50, start_time, time_offset)
+
+time_offset = Period("T0.1S")
+t5 = create_random_table(5, start_time, time_offset)
+t6 = create_random_table(50, start_time, time_offset)
```
Comment (jjbrosnan) [inline demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:59]: Generally speaking, you want to use DynamicTableWriter in a thread, unless you want the console to lock up while a table is being written using DynamicTableWriter.  I know these only write small tables with no wait time between new rows, so it creates the tables pretty instantaneously, but I'd still recommend using threads.  You could also show concurrency.

Also, this notebook only shows how to use DynamicTableWriter.  What about other methods to create tables like timeTable, Replayer, etc?

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md (hunk 5 above)
Comment (jjbrosnan) [inline demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:62]: In the first line, you mention doing operations on tables, but don't do any here.  I'd reword the start and remove mention of doing operations on data when they're not present in the notebook.

**Review by jjbrosnan — COMMENTED** [review]
General feedback regarding scope and code

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jakemulf) [reply demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:3]: The way Pete laid it out was that this is supposed to be part of a "real-time table ops" series, with this being the first entry. I think the intro Margaret has will make more sense with the context of the entire series

[the remark this answers is not in the document]
Comment (jakemulf) [reply demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:3]: How does this sound?

```
Throughout this demo notebook series, we show many of the ways to interact with real-time data in Deephaven. Here, we create some tables with fake data; in other notebooks, we show how to perform table operations on that data. Knowing how to create fake ticking tables can be useful for familiarizing yourself with Deephaven and to work on proof of concepts without necessarily having a complete dataset.
```

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md (hunk 5 above)
Comment (jakemulf) [reply demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:62]: That is correct, this is a placeholder. I'm assuming we're gonna publish this entire folder at once so this won't be a dead link on the live demo

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md (hunk 7 above)
Comment (jakemulf) [reply demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:59]: Ooh that's good to know, did not know that about dynamic table writer. I'll update this to thread the method

I felt that dynamic table writer was the most intuitive way to create random/fake data at scale. timeTable could certainly work, but that depends on running it for a certain amount of time, and Replayer requires an already existing table (right?). We could use emptyTable and methods to generate random data, something like

```
from deephaven.TableTools import emptyTable

import random

def random_int():
    return random.randint(1,100)

result = emptyTable(5)

result = result.update("Numbers = random_int()")
```

Would that be preferable? Or good to demo as an alternative to dynamic table writer?

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md (hunk 5 above)
Comment (jakemulf) [reply demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:62]: Let's move discussion to https://github.com/deephaven/deephaven-core/pull/1846#discussion_r787697946 since that's pretty similar to this comment, this might make more sense in the context of the entire folder

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md [hunk 8]
```diff
@@ -0,0 +1,68 @@
+# Create tables and fake data
+
+Throughout this demo notebook series, we show many of the ways to interact with real-time data in Deephaven. Here, we create some tables with fake data; in other notebooks, we show how to perform table operations on that data. Knowing how to create fake ticking tables can be useful for familiarizing yourself with Deephaven and to work on proof of concepts without necessarily having a complete dataset.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:3]:
```suggestion
Throughout this demo notebook series, we show many of the ways to interact with real-time data in Deephaven. Here, we create some tables with fake data; in other notebooks, we show how to perform table operations on that data. Knowing how to create fake ticking tables is useful for familiarizing yourself with Deephaven, but also for working on proof of concepts without necessarily having a complete dataset.
```

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md [hunk 9]
```diff
@@ -0,0 +1,68 @@
+# Create tables and fake data
+
+Throughout this demo notebook series, we show many of the ways to interact with real-time data in Deephaven. Here, we create some tables with fake data; in other notebooks, we show how to perform table operations on that data. Knowing how to create fake ticking tables is useful for familiarizing yourself with Deephaven, but also for working on proof of concepts without necessarily having a complete dataset.
```
Comment (jjbrosnan) [inline demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md:3]: Nice intro

**Review by jjbrosnan — APPROVED** [review]
Looks great after the updates!

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md (hunk 6 above)
[the remark this answers is not in the document]
Comment (margaretkennedy) [reply demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md]: LGTM

File: demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md (hunk 7 above)
Comment (margaretkennedy) [reply demo/web/src/main/notebooks/A. Real-time table ops/A1 Create tables and fake data .md]: @jjbrosnan

**Review by margaretkennedy — APPROVED** [review]

File: demo/web/src/main/notebooks/A. Real-time table ops/A2 Filter and decorate .md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+daily_data = create_random_table(365, start_time, time_offset)
+```
+
+Now we decorate the data by adding its day of the week.
+
+```python
+daily_data = daily_data.update("DayOfWeekInt = dayOfWeek(DateTime, TZ_NY)")
+```
+
+Next,  we convert the day of week to a string representation.
```
Comment (margaretkennedy) [inline demo/web/src/main/notebooks/A. Real-time table ops/A2 Filter and decorate .md:20]:
```suggestion
Next, we convert the day of week to a string representation.
```

**Review by margaretkennedy — APPROVED** [review]

File: demo/web/src/main/notebooks/A. Real-time table ops/A3 Do time series and relational joins.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+Let's join these tables using a `raj`.
+
+```python
+joined_data_raj = daily_data_0.raj(daily_data_1, "DateTime", "Number1 = Number, Character1 = Character, Boolean1 = Boolean")
+```
+
+And now we have our table joined as expected.
+
+[The next notebook](A4%20Group%20and%20aggregate.md) will show how to group and aggregate this data.
```
Comment (jakemulf) [inline demo/web/src/main/notebooks/A. Real-time table ops/A3 Do time series and relational joins.md:52]:
```suggestion
```
